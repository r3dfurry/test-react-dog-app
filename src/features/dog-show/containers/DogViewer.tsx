import * as React from 'react';
import { loadListOfDogs, loadSpecificDog } from '../actions';
import { connect } from 'react-redux';
import { ModeSwitcher } from '../components/mode-switcher';
import ImageShow from '../components/image-show';
import * as Immutable from 'immutable';

interface IProps {
    breeds: Immutable.List<string>;
    imageLink: string;
    selectedBreed: string;
    loadListOfDogs: () => {};
    loadSpecificDog: (breed: string) => {};
}
 
class DogViewer extends React.Component<IProps, any> {
    constructor(props: any) {
        super(props);
    }
    public componentWillMount() {
        this.props.loadListOfDogs();
    }
    public render() {
        return (
            <div>
                <ModeSwitcher
                    breeds={this.props.breeds}
                    selectedBreed={this.props.selectedBreed}
                    showRandomBreedClick={this.props.loadListOfDogs}
                    showSpecificBreedClick={this.props.loadSpecificDog}/>
                <ImageShow
                    breed={this.props.selectedBreed}
                    imageLink={this.props.imageLink} />
            </div>
        )
    }
}
const mapStateToProps = (state: any, ownProps: any) => {
    // tslint:disable-next-line:no-console
    console.log(state.specifyDog.get('images'));
    return {
        breeds: state.listOfDogs.get('breeds'),
        imageLink: state.specifyDog.get('images').get(0),
        isFetching: state.listOfDogs.get('isFetching'),
        selectedBreed: state.listOfDogs.get('selectedBreed')
    };
}

export default connect(mapStateToProps, {
    loadListOfDogs,
    loadSpecificDog
})(DogViewer);