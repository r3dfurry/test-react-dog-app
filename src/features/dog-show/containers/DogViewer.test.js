import * as React from 'react';
import DogViewer from './DogViewer';
import { Provider } from 'react-redux'; 
import { create } from 'react-test-renderer';
import configureStore from '../store';

test('1', () => {
    const store = configureStore();

    const component = create(
      <Provider store={store}>
        <DogViewer />
      </Provider>
    ).toJSON();
    
    expect(component).toMatchSnapshot();

});
