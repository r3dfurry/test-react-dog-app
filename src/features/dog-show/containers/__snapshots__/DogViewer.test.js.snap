// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 1`] = `
<div>
  <div>
    <div>
      <button
        className="ui labeled icon button"
        onClick={[Function]}
      >
        <i
          className="random icon"
        />
        Show random breed
      </button>
      <span>
        or 
        <br />
        specify breed
      </span>
      <select
        className="ui dropdown"
        onChange={[Function]}
        value={undefined}
      >
        <option
          value=""
        >
          
        </option>
      </select>
    </div>
  </div>
</div>
`;
